Non-scaling SVG scatter points

rreusser opened the pull request:
See issue: #760
See related line scaling PR: #761 
See codepen example (visible on scrollZoom): http://codepen.io/rsreusser/pen/OXvmjR?editors=0010

This PR compensates for point scaling by applying an inverse scale transform. It's simple, but kind of an ugly hack since it requires stripping any existing scale from points before applying a new scale. That may make it prohibitive, but it was a bit easier than expected, so maybe it's not unreasonable. cc @mdtusz

rreusser commented on src/lib/index.js:
```diff
@@ -484,6 +484,27 @@ lib.setScale = function(element, x, y) {
     return transform;
 };
 
+lib.setPointScale = function(selection, x, y) {
```
This is scatter-specific. It's here since related functions are right next to it, but it probably should not in `lib/`.

etpinard replied:
I'm ok with this being in `lib/`. Putting this in `src/traces/scatter/set_point_scale.js` would be ok too. Your call @rreusser

etpinard commented on src/plots/cartesian/dragbox.js:
```diff
@@ -633,7 +633,10 @@ module.exports = function dragBox(gd, plotinfo, x, y, w, h, ns, ew) {
                 .call(Lib.setTranslate, plotDx, plotDy)
-                .call(Lib.setScale, xScaleFactor, yScaleFactor);
+                .call(Lib.setScale, xScaleFactor, yScaleFactor)
+                .selectAll('.points').selectAll('.point')
```
a one-line comment here would be nice.

etpinard commented:
Nice solution. Looks like isolating the `setPointScale` call in `dragbox.js` worked out well.

@rreusser Would you mind adding a few unit tests for `setPointScale` ?

rreusser commented:
@etpinard Should `setPointScale` work with d3 elements, like `setTranslate`? As long as it's clearly implemented, tested, and commented, it seems unnecessary to implement at the moment unless needed.

etpinard commented:
> Should setPointScale work with d3 elements, like setTranslate?

No. I don't see the need.

rreusser commented:
Added comment, changed the name to be more descriptive, added tests, removed scale if = `(1, 1)`, and confirmed that it still has the same visual effect.

etpinard commented:
Very nicely done. :dancer: merge away.
